### PR TITLE
Improve matmul benchmark

### PR DESCRIPTION
**Pull request by sayandip18**
This is WIP. More commits incoming.

### CODE REVIEW

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Maybe you could parametrize things instead of using multiple separate benchmarks?

**Comment by sayandip18** [comment]
If the changes are okay, I'll squash the commits

File: benchmarks/benchmark_matmul.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import numpy as np
+import sparse
+
+
+class Matmul_Sparse:
```
Comment (hameerabbasi) [inline benchmarks/benchmark_matmul.py:5]: Maybe these two classes could also be parameterised.

Something like `dense_arg = [0, 1, None]`.

**Review by hameerabbasi — COMMENTED** [review]
I'd like one last change, parametrise all the things!

File: benchmarks/benchmark_matmul.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    params = (["coo", "gcxs"], [0.01, 0.33, 0.5, 1.0])
+
+    def setup(self, p, dens_arg):
+        np.random.seed(0)
+        self.x = sparse.random((100, 100), density=0.01, format=p)
+        self.y = sparse.random((100, 100), density=dens_arg, format=p)
```
Comment (hameerabbasi) [inline benchmarks/benchmark_matmul.py:11]: This still produces a GCXS when the dense argument is `1.0`, what I meant in https://github.com/pydata/sparse/pull/467/files#r633073348 was that `0` means `self.x` should be `.todense()`, and `1` means the same for `self.y`.

File: benchmarks/benchmark_matmul.py [hunk 3]
```diff
@@ -3,12 +3,17 @@
...
-    params = (["coo", "gcxs"], [0.01, 0.33, 0.5, 1.0])
+    params = (["coo", "gcxs"], [0, 1, None])
 
     def setup(self, p, dens_arg):
         np.random.seed(0)
         self.x = sparse.random((100, 100), density=0.01, format=p)
-        self.y = sparse.random((100, 100), density=dens_arg, format=p)
+        self.y = sparse.random((100, 100), density=0.01, format=p)
+
+        if(dens_arg==1):
```
Comment (hameerabbasi) [inline benchmarks/benchmark_matmul.py:13]:
```suggestion
        if(dens_arg==0):
```

File: benchmarks/benchmark_matmul.py [hunk 4]
```diff
@@ -3,12 +3,17 @@
...
+            self.x = self.x.todense()
+        elif(dens_arg==0):
```
Comment (hameerabbasi) [inline benchmarks/benchmark_matmul.py:15]:
```suggestion
        elif(dens_arg==1):
```

**Review by hameerabbasi — APPROVED** [review]
Just waiting on CI to turn green.

**Pull request merged**